src/runtime/set-value.ts

```diff
@@ -11,6 +11,27 @@ export const getValue = (ref: d.RuntimeRef, propName: string) => getHostRef(ref)
 export const setValue = (ref: d.RuntimeRef, propName: string, newVal: any, cmpMeta: d.ComponentRuntimeMeta) => {
   // check our new property value against our internal value
   const hostRef = getHostRef(ref);
+
+  /**
+   * If the host element is not found, let's fail with a better error message and provide
+   * details on why this may happen. In certain cases, e.g. see https://github.com/ionic-team/stencil/issues/5457,
+   * users might import a component through e.g. a loader script, which causes confusions in runtime
+   * as there are multiple runtimes being loaded and/or different components used with different
+   * loading strategies, e.g. lazy vs implicitly loaded.
+   *
+   * Todo(@christian-bromann): remove, once a solution for this was identified in https://outsystemsrd.atlassian.net/browse/STENCIL-1308 and implemented
```

Comment:
we might want to just do `TODO(STENCIL-1308): $MESSAGE` to align with our other TODO comments scattered around the codebase